views-react/src/main/java/io/micronaut/views/react/ReactViewsRenderer.java

```diff
@@ -60,9 +63,14 @@ class ReactViewsRenderer<PROPS> implements ViewsRenderer<PROPS, HttpRequest<?>>
     @Override
     public @NonNull Writable render(@NonNull String viewName, @Nullable PROPS props, @Nullable HttpRequest<?> request) {
         return writer -> {
+            var renderCallback = new RenderCallback(writer, request);
+
             try {
                 beanPool.useContext(handle -> {
-                    render(viewName, props, writer, handle.get(), request);
+                    render(viewName, props, renderCallback, handle.get(), request);
+                    do {
+                        LOG.trace("Waiting for render callback to complete...");
+                    } while (!renderCallback.isWriteCompleted());
```

Comment:
How does this work? It looks like a tight loop reading from a non-volatile field? I don't understand the intended control flow here.

I think the original feature request is asking for the ability for views to return a Publisher so view rendering can be chunked alongside other async operations like DB calls, it's not something specific to React.

Reply:
Yes we can't have this loop, it needs to be a blocking check like a CompletableFuture or something.

To get full Publisher support for views renderers would be too far though imo (and wouldn't automatically imply chunked writing)

Reply:
Fortunately the original request doesn't ask for chunked writing, it wouldn't be needed in this context anyway.

Reply:
Please, @mikehearn and @yawkat review the PR again. I have added a reactive API.